scripts/west_commands/sign.py

```diff
@@ -633,3 +651,120 @@ def sign(self, command, build_dir, build_conf, formats):
 
         os.remove(out_bin)
         os.rename(out_tmp, out_bin)
+
+class SirpsSigner(Signer):
+    """Signer class for Silicon Labs Commander tool."""
+
+    def sign(self, command, build_dir, build_conf, formats):
+        """Sign the Zephyr binary using Silicon Labs Commander.
+
+        :param command: The Sign instance (provides args and utility methods)
+        :param build_dir: The build directory path
+        :param build_conf: BuildConfiguration object for the build directory
+        :param formats: List of formats to generate (e.g., ['bin', 'rps'])
+        """
+        self.command = command
+        args = command.args
+        b = pathlib.Path(build_dir)
+
+        # Check if signing is needed
+        if not formats:
+            if not args.quiet:
+                command.inf("No output formats specified, skipping signing.")
+            return
+
+        # Get tool path from args or default to system PATH
+        tool_path = args.tool_path if args.tool_path else shutil.which("commander")
+        if not tool_path:
+            command.die("commander not found; either install it or provide --tool-path")
+
+        if not args.quiet:
+            command.inf(f"Signing with tool {tool_path}")
+
+        # Load CMake cache to get BOARD_DIR
+        cache = CMakeCache.from_build_dir(build_dir)
+
+        # Get kernel binary name (default to 'zephyr')
+        kernel_name = build_conf.get('CONFIG_KERNEL_BIN_NAME', 'zephyr')
+
+        # Input and output files
+        in_bin = b / 'zephyr' / f"{kernel_name}.bin.rps"  # Raw binary input
+        out_rps = args.sbin or (b / 'zephyr' / f"{kernel_name}.bin.rps")  # Signed output
+
+        # Check if input binary exists
+        if not in_bin.is_file():
+            command.die(f"No unsigned .bin found at {in_bin}")
+
+        # Use $ENV{PWD} as working directory
+        working_dir = os.environ.get('PWD')
+        if not working_dir:
+            command.die("PWD environment variable not set")
+
+        # Get signing parameters with board Kconfig as primary source
+        m4_ota_key = None
+        m4_private_key = None
+
+        # Primary source: Board Kconfig defaults
+        board_dir = cache.get('BOARD_DIR') # Fallback to PWD if BOARD_DIR missing
+        board_kconfig = pathlib.Path(board_dir) / 'Kconfig.siwx917_r4338a'
```

Comment:
You can't hard-code the board Kconfig path, this must generically work for all SiWx91x based boards.